src/bridge/cockpitpackages.c

```diff
@@ -338,30 +339,26 @@ package_walk_directory (GChecksum *own_checksum,
 }
 
 static JsonObject *
-read_json_file (const gchar *directory,
-                const gchar *name,
-                GError **error)
+read_json_file (const gchar *path,
+                GError **error,
+                GChecksum *checksum)
 {
-  JsonObject *object;
-  GMappedFile *mapped;
-  gchar *filename;
-  GBytes *bytes;
-
-  filename = g_build_filename (directory, name, NULL);
-  mapped = g_mapped_file_new (filename, FALSE, error);
-  g_free (filename);
+  g_autoptr(GMappedFile) mapped = g_mapped_file_new (path, FALSE, error);
 
   if (!mapped)
+    return NULL;
+
+  g_autoptr(GBytes) bytes = g_mapped_file_get_bytes (mapped);
+
+  if (checksum)
     {
-      return NULL;
+      gsize contents_size;
+      const void *contents = g_bytes_get_data (bytes, &contents_size);
+      if (contents)
+        g_checksum_update (checksum, contents, contents_size);
```

Comment:
That's a bit different than the existing code does it.  It also hashes the filename, along with the content, and uses nuls to separate things.  I have trouble thinking about a case where this would make a difference, however.  On the other hand, it's also a bit uncouth (imho) to add +1 to the size you get out of a GBytes and assume that it's definitely nul terminated...

Reply:
I liked this better back then, but with the new approach it's back to the string checksum.